packages/atomic/src/components/commerce/atomic-product-excerpt/atomic-product-excerpt.ts

```diff
@@ -0,0 +1,148 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import type {Product} from '@coveo/headless/commerce';
+import {html, LitElement, nothing, unsafeCSS} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {booleanConverter} from '@/src/converters/boolean-converter';
+import {bindingGuard} from '@/src/decorators/binding-guard';
+import {bindings} from '@/src/decorators/bindings';
+import {createProductContextController} from '@/src/decorators/commerce/product-template-decorators';
+import {errorGuard} from '@/src/decorators/error-guard';
+import {injectStylesForNoShadowDOM} from '@/src/decorators/light-dom';
+import type {InitializableComponent} from '@/src/decorators/types';
+import {withTailwindStyles} from '@/src/decorators/with-tailwind-styles';
+import {
+  renderExpandableText,
+  type TruncateAfter,
+} from '../../common/expandable-text/expandable-text';
+import type {CommerceBindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+import styles from './atomic-product-excerpt.tw.css';
+
+/**
+ * The `atomic-product-excerpt` component renders the excerpt of a product.
+ *
+ * @part expandable-text - The part that contains the product excerpt text.
+ * @alpha
+ */
+@customElement('atomic-product-excerpt')
+@bindings()
+@injectStylesForNoShadowDOM
+@withTailwindStyles
+export class AtomicProductExcerpt
+  extends LitElement
+  implements InitializableComponent<CommerceBindings>
+{
+  static styles = unsafeCSS(styles);
+  @state()
+  bindings!: CommerceBindings;
+
+  public error!: Error;
+
+  private productController = createProductContextController(this);
+
+  @state() private isExpanded = false;
+  @state() private isTruncated = false;
+
+  private excerptText!: HTMLDivElement;
+  private resizeObserver: ResizeObserver;
+  /**
+   * The number of lines after which the product excerpt should be truncated. A value of "none" will disable truncation.
+   */
+  @property({type: String, attribute: 'truncate-after'})
+  public truncateAfter: TruncateAfter = '2';
+
+  /**
+   * Whether the excerpt should be collapsible after being expanded.
+   */
+  @property({
+    type: Boolean,
+    attribute: 'is-collapsible',
+    converter: booleanConverter,
+  })
+  public isCollapsible = false;
+
+  public initialize() {
+    this.validateProps();
+  }
+
+  constructor() {
+    super();
+    this.resizeObserver = new ResizeObserver(() => {
+      if (
+        this.excerptText &&
+        this.excerptText.scrollHeight > this.excerptText.offsetHeight
+      ) {
+        this.isTruncated = true;
+      } else {
+        this.isTruncated = false;
+      }
```

Comment:
Really :D ?
```suggestion
        this.isTruncated = this.excerptText && this.excerptText.scrollHeight > this.excerptText.offsetHeight;
```
(unlinted suggestion)